lib/client_summarizer.rb

```diff
@@ -0,0 +1,31 @@
+class ClientSummarizer
+  include FiscalYearMixin
+
+  attr_reader :client_namespace, :fiscal_year
+
+  def initialize(args)
+    @client_namespace = args[:client_namespace]
+    now = Time.zone.now
+    @fiscal_year = (args[:fiscal_year] || self.class.which_fiscal_year(now.year, now.month)).to_i
+    @_records = build_records
```

Comment:
I've been following this discussion re: underscore for memoization: https://github.com/thoughtbot/guides/pull/373

historically I only see underscored variables in Ruby in order to indicate that a variable is unused. This is enforced via rubocop. example:

```ruby
{ 1: "hi", 2: "hello" }.each do |_key, value|
  puts value
end
```

I am fine with doing underscores for memoized ivars in this project. In cases like this one, I'd probably not use an underscore.